src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -803,6 +812,18 @@ public async Task VersionHandshakeAsync(NetworkPeerRequirement requirements, Can
                 // Ask the just-handshaked peer for the peers they know about to aid in our own peer discovery.
                 await this.SendMessageAsync(new GetAddrPayload(), cancellationToken).ConfigureAwait(false);
             }
+            catch (OperationCanceledException e)
+            {
+                this.logger.LogDebug("Operation canceled during handshake.");
```

Comment:
Do we need to `throw` inside these new catches?

Reply:
Yes, fixed